modules/subprocess/src/main/scala/notebook/kernel/pfork/BetterFork.scala

```diff
@@ -170,7 +169,7 @@ object BetterFork {
         } else {
           acc ++ (cl.asInstanceOf[URLClassLoader].getURLs map { u =>
             val f = new File(u.getFile)
-            URLDecoder.decode(f.getAbsolutePath, "UTF8")
+            URLDecoder.decode(f.getParent, "UTF8")
```

Comment:
You're right, that is too aggressive. I only had the spark-notebook libs in mind, since they all need to be included, one by one, I just decided to take'em all. On the other side, a lot of java/ext classes will be added to the CP, that shouldn't. I guess it could be a resource waster and potentially cause conflicts with versions. The former wouldn't hurt I guess, and the latter would be unfortunate anyway, as one should build against the JVM version that one will be running. Could something else happen?